chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -107,20 +107,26 @@ impl EthereumAdapter {
     ) -> Self {
         let web3 = Arc::new(Web3::new(transport));
 
-        info!(logger, "Checking if provider supports getBlockReceipts"; "provider" => &provider);
+        info!(logger, "Checking if provider supports getBlockReceipts");
 
         // Check if the provider supports `getBlockReceipts` method.
-        let supports_block_receipts = Self::check_block_receipt_support(
-            web3.clone(),
-            &provider,
-            supports_eip_1898,
-            call_only,
-            &logger,
-        )
-        .await
-        .is_ok();
-
-        info!(logger, "Checked if provider supports eth_getBlockReceipts"; "provider" => &provider, "supports_block_receipts" => supports_block_receipts);
+        let block_receipts_support_result =
+            Self::check_block_receipt_support(web3.clone(), supports_eip_1898, call_only, &logger)
+                .await;
+
+        // Error message to log if the check fails.
+        let error_message = block_receipts_support_result
```

Comment:
this here seems unnecessarily verbose, printing stuff like "call only providers no supported" is not really something we care about and may cause operators to be confused. Maybe a simple thing would be to have `check_block_receipt_support` return Result<bool>, this if it's just a "no" then we don't need to print anything, if there's an actual error we can print it. Alternatively we can define the error (instead of anyhow) and for specific types of errors we can print, although my gut feeling is that you will prolly end up with `enum Error { Unsupported, Unknown(#[from] anyhow::Error) ` which may not be too useful

Reply:
Good point that was the case in my initial commit, but decided against it since we might be interested in the actual error message? If the operator sees that a provider which used to support block receipts stopped supporting it, the would be interested in knowing the reason to report to the provider, without manually reproducing? Wdyt?

Reply:
maybe what we should do is have sort of endpoint on the status server that shows the status of these things? It could be useful for other stuff as well as these seem to be stateful decisions made once and kept for the duration of the process. I think could be very useful for the status of endpoints as well as genesis validation which is coming soon. I think logs might not be the right medium for this type of information

Reply:
Ah, yes that sounds better. That would probably mean us needing store this state somewhere in the DB right?
Especially when we have a multi graph-node setup